src/transformers/modeling_bart.py

```diff
@@ -1090,7 +1090,7 @@ def forward(
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
         if labels is not None:
-            assert -100 not in labels
+            assert labels.min() > 0, f'negative labels are not supported, got {labels.min()}'
```

Comment:
This should be `labels.min()>=0`, right?

Reply:
yes good catch

Reply:
I don't think this is something we can accept in the forward pass of the BART model, as it would severely harm TPU performance. The assertion means that we would be retrieving the value of the xla tensor `labels.min() > 0` back on CPU every time, which would cause a big performance drop.

I would advocate for this to be put in the dataloader instead, and the loss will crash anyway when seeing a label value which has a negative value and is not the ignored index.

Reply:
Arg,  we completely ignored TPU, makes sense. Thanks Lysandre!